### PR TITLE
use simpler java collection instead of array in internal of Poller class

### PR DESCRIPTION
**Pull request by esahekmat**
it's more readable and simpler to work with List instead of an array. it removes all the resizing and size management stuff

### CODE REVIEW

File: src/main/java/org/zeromq/ZMQ.java [hunk 1]
```diff
@@ -3637,11 +3633,8 @@ protected Poller(Context context, int size)
             selector = context.selector();
             assert (selector != null);
 
-            items = new PollItem[size];
+            items = new LinkedList<>();
```
Comment (fbacchella) [inline src/main/java/org/zeromq/ZMQ.java]: As size is now useless, the constructor using a size should be deprecated. Or size could be used when creating the list, it avoid some useless resizing.

**Comment by colini** [comment]
I'm wondering why LinkedList instead of ArrayList?

**Comment by fbacchella** [comment]
If you add and remove many elements, it will do less reallocation of full array. But as the real key for search is the socket, not the index (a legacy from C code probably), I wonder if using a map could be better ? With a deprecation added for all index-based access.

File: src/main/java/org/zeromq/ZMQ.java (hunk 1 above)
Comment (esahekmat) [reply src/main/java/org/zeromq/ZMQ.java]: resizing is cheap on LinkedList, but you're right, I should deprecate this constructor and create a new one without size

**Comment by esahekmat** [comment]
exactly, I chose LinkedList because its resizing is cheap, map sounds good, it's faster for search which is the used frequently

**Comment by esahekmat** [comment]
Actually, I found out that methods like poller.pollout(index) and poller.pollin(index) is more frequently used over and over usually in a while loop, and there are no similar methods getting Socket instead of index, I can add these methods and use a map instead of LinkedList the only problem is Poller accept SelectableChannel also, and we should use Object as the key of the map.

**Comment by trevorbernard** [comment]
Subjectively, I like array backed data structures more because they're more cache friendly because you aren't jumping from pointer to pointer especially in a hot loop. But in any case, it's a good commit so it will be merged.

**Comment by esahekmat** [comment]
Thank you @trevorbernard, from your comment I understand that it's better to use ArrayList to avoid jumping from pointer to pointer. so let me open another pull request in order to change LinkedList to ArrayList

**Pull request merged**